hudi-common/src/main/java/org/apache/hudi/common/table/read/FileGroupRecordBuffer.java

```diff
@@ -275,43 +274,47 @@ protected Option<Pair<Option<T>, Map<String, Object>>> doProcessNextDataRecord(T
       } else {
         switch (recordMergeMode) {
           case COMMIT_TIME_ORDERING:
-            return Option.empty();
+            return Option.of(Pair.of(Option.ofNullable(record), metadata));
```

Comment:
how come we had such a basic bug wrt custom merger :(

Reply:
can we add tests for this case please?

Reply:
The existing tests already can cover.